code/game/objects/items/weapons/RCD.dm

```diff
@@ -382,3 +389,285 @@ RCD
+	set name = "Choose Light Color"
+	set category = "Object"
+	color_choice = input(usr,"Choose Color") as color
+
+
+/obj/item/weapon/rld/New()
+	..()
+
+	desc = "An RLD. It currently holds [matter]/[max_matter] matter-units."
+	src.spark_system = new /datum/effect_system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+
+
+/obj/item/weapon/rld/Destroy()
+	qdel(spark_system)
+	spark_system = null
+	. = ..()
+
+/obj/item/weapon/rld/attackby(obj/item/weapon/W, mob/user, params)
+	var/loaded = 0
+	if(istype(W, /obj/item/weapon/rcd_ammo))
+		var/obj/item/weapon/rcd_ammo/R = W
+		if((matter + R.ammoamt) > max_matter)
+			to_chat(user, "<span class='warning'>The RLD can't hold any more matter-units!</span>")
+			return
+		qdel(W)
+		matter += R.ammoamt
+		icon_state = "rld-[round(matter/35)]"
+		update_icon()
+		playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		loaded = 1
+	else if(istype(W, /obj/item/stack/sheet/glass))
+		loaded = loadwithsheets(W, sheetmultiplier, user)
+	if(loaded)
+		to_chat(user, "<span class='notice'>The RLD now holds [matter]/[max_matter] matter-units.</span>")
+		desc = "An RLD. It currently holds [matter]/[max_matter] matter-units."
+	else
+		return ..()
+
+/obj/item/weapon/rld/proc/loadwithsheets(obj/item/stack/sheet/S, value, mob/user)
+	var/maxsheets = round((max_matter-matter)/value)    //calculate the max number of sheets that will fit in RLD
+	if(maxsheets > 0)
+		var/amount_to_use = min(S.amount, maxsheets)
+		S.use(amount_to_use)
+		matter += value*amount_to_use
+		playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You insert [amount_to_use] [S.name] sheets into the RLD. </span>")
+		icon_state = "rld-[round(matter/35)]"
+		update_icon()
+		return 1
+	to_chat(user, "<span class='warning'>You can't insert any more [S.name] sheets into the RLD!")
+	return 0
+
+/obj/item/weapon/rld/attack_self(mob/user)
+	//Change the mode
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+	switch(mode)
+		if(1)
+			mode = 2
+			to_chat(user, "<span class='notice'>You change RLD's mode to 'Permanent Light Construction'.</span>")
+		if(2)
+			mode = 3
+			to_chat(user, "<span class='notice'>You change RLD's mode to 'Light Launcher'.</span>")
+		if(3)
+			mode = 1
+			to_chat(user, "<span class='notice'>You change RLD's mode to 'Deconstruct'.</span>")
+	if(prob(20))
+		src.spark_system.start()
+
+/obj/item/weapon/rld/proc/activate(atom/A)
+	playsound(src.loc, 'sound/items/Deconstruct.ogg', 50, 1)
+
+/obj/item/weapon/rld/proc/checkdupes(var/target)
+	. = list()
+	var/turf/checking = get_turf(target)
+	for(var/obj/machinery/light/dupe in checking)
+		if(istype(dupe, /obj/machinery/light))
+			. |= dupe
+
+
+/obj/item/weapon/rld/afterattack(atom/A, mob/user)
+	var/turf/start = get_turf(src)
+	if(!(A in view(7, get_turf(start))))
+		to_chat(user, "<span class='warning'>The \'Out of Range\' light on the RLD blinks red.</span>")
+		return 0
+	switch(mode)
+		if(1)
```

Comment:
Goof you're gonna have to explain how I can make this an effective subtype of RCD's when they do very different things. Otherwise im just going to do what Antur suggested.